servers/cromwell/jobs/controllers/utils/auth.py

```diff
@@ -0,0 +1,29 @@
+from flask import current_app, request
+from requests.auth import HTTPBasicAuth
+from werkzeug.exceptions import Unauthorized
+
+
+def requires_auth(fn):
```

Comment:
I like this! Could you test using mock request and either decorating a function in your tests or else calling requires_auth directly?